apps/api/src/database/repositories/leaderboard-repository.ts

```diff
@@ -109,6 +112,142 @@ async function getGlobalStatsImpl(type: CompetitionType): Promise<{
+    .from(competitions)
+    .where(eq(competitions.status, "ended"));
+
+  if (allEndedCompetitions.length === 0) {
+    return {
+      activeAgents: 0,
+      totalTrades: 0,
+      totalPositions: 0,
+      totalVolume: 0,
+      totalCompetitions: 0,
+      totalVotes: 0,
+      competitionIds: [],
+    };
+  }
+
+  // Separate competition IDs by type for efficient querying
+  const paperTradingIds = allEndedCompetitions
+    .filter((c) => c.type === "trading")
+    .map((c) => c.id);
+  const perpsIds = allEndedCompetitions
+    .filter((c) => c.type === "perpetual_futures")
+    .map((c) => c.id);
+  const allCompetitionIds = allEndedCompetitions.map((c) => c.id);
+
+  // Run all queries in parallel for maximum efficiency
+  const [
+    tradeStats,
+    positionStats,
+    perpsVolumeStats,
+    voteStats,
+    activeAgentStats,
+  ] = await Promise.all([
+    // 1. Get trade stats for paper trading competitions only
+    paperTradingIds.length > 0
+      ? dbRead
+          .select({
+            totalTrades: drizzleCount(trades.id),
+            totalVolume: sum(trades.tradeAmountUsd).mapWith(Number),
+          })
+          .from(trades)
+          .where(inArray(trades.competitionId, paperTradingIds))
+      : Promise.resolve([{ totalTrades: 0, totalVolume: 0 }]),
+
+    // 2. Get position stats for perps competitions only
+    perpsIds.length > 0
+      ? dbRead
+          .select({
+            totalPositions: drizzleCount(perpetualPositions.id),
+          })
+          .from(perpetualPositions)
+          .where(inArray(perpetualPositions.competitionId, perpsIds))
+      : Promise.resolve([{ totalPositions: 0 }]),
+
+    // 3. Get volume stats for perps competitions
+    perpsIds.length > 0
+      ? dbRead
+          .selectDistinctOn([perpsAccountSummaries.agentId], {
+            totalVolume: perpsAccountSummaries.totalVolume,
+          })
+          .from(perpsAccountSummaries)
+          .where(inArray(perpsAccountSummaries.competitionId, perpsIds))
+          .orderBy(
+            perpsAccountSummaries.agentId,
+            desc(perpsAccountSummaries.timestamp),
+          )
+      : Promise.resolve([]),
+
+    // 4. Get vote stats for all competitions
+    dbRead
+      .select({
+        totalVotes: drizzleCount(votes.id),
+      })
+      .from(votes)
+      .where(inArray(votes.competitionId, allCompetitionIds)),
+
+    // 5. Get active agent count across all competitions
+    dbRead
+      .select({
+        totalActiveAgents: countDistinct(competitionAgents.agentId),
+      })
+      .from(competitionAgents)
+      .where(
+        and(
+          inArray(competitionAgents.competitionId, allCompetitionIds),
+          eq(competitionAgents.status, "active"),
+        ),
+      ),
+  ]);
+
+  // Calculate total perps volume from the latest account summaries
+  const perpsVolume = perpsVolumeStats.reduce((total, summary) => {
+    const volume = Number(summary.totalVolume) || 0;
+    return total + volume;
+  }, 0);
```

Comment:
Should we use a SQL `SUM()` instead so that we don't have to calculate this here in memory?

Reply:
Nice - already addressed as per one of @joewagner 's comment above